### PR TITLE
fix(sheets-data-validation): date validator error when errorStyle=STOP was set

### PR DESCRIPTION
**Pull request by weird94**
close https://github.com/dream-num/univer/issues/3098

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/data-validation/src/models/data-validation-model.ts [hunk 1]
```diff
@@ -161,10 +161,10 @@ export class DataValidationModel<T extends IDataValidationRule = IDataValidation
         return manager.getDataValidations();
     }
 
-    validator(content: Nullable<CellValue>, rule: T, pos: any) {
+    validator(rule: T, pos: any, value: any) {
```
Comment (wzhudev) [inline packages/data-validation/src/models/data-validation-model.ts:164]: Same here.

**Review by wzhudev — CHANGES_REQUESTED** [review]

File: packages/data-validation/src/models/data-validation-manager.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export class DataValidationManager<T extends IDataValidationRule> extends Dispos
         return this._dataValidations;
     }
 
-    validator(_content: Nullable<CellValue>, _rule: IDataValidationRule, _pos: any, _onComplete: (status: DataValidationStatus, changed: boolean) => void) {
+    validator(_value: any, _rule: IDataValidationRule, _pos: any, _onComplete: (status: DataValidationStatus, changed: boolean) => void) {
```
[the remark this answers is not in the document]
Comment (weird94) [reply packages/data-validation/src/models/data-validation-manager.ts]: fixed

File: packages/data-validation/src/models/data-validation-model.ts (hunk 1 above)
Comment (weird94) [reply packages/data-validation/src/models/data-validation-model.ts]: fixed

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**